components/builder-api/src/server/resources/channels.rs

```diff
@@ -14,371 +14,326 @@
-                        Ok(_) => match audit_package_rank_change(
-                            req,
-                            package.get_id(),
-                            origin_channel.get_id(),
-                            PackageChannelOperation::Demote,
-                            origin_id,
-                        ) {
-                            Ok(_) => Ok(Response::with(status::Ok)),
-                            Err(err) => return Ok(render_net_error(&err)),
-                        },
-                        Err(err) => return Ok(render_net_error(&err)),
-                    }
+                if channel == "stable" || channel == "unstable" {
+                    return HttpResponse::new(StatusCode::FORBIDDEN);
+                }
+
+                let mut delete = OriginChannelDelete::new();
+                delete.set_id(origin_channel.get_id());
+                delete.set_origin_id(origin_channel.get_origin_id());
+                match route_message::<OriginChannelDelete, NetOk>(req, &delete) {
+                    Ok(_) => HttpResponse::new(StatusCode::OK),
+                    Err(err) => err.into(),
                 }
-                Err(err) => Ok(render_net_error(&err)),
             }
+            Err(err) => err.into(),
         }
-        Err(err) => Ok(render_net_error(&err)),
     }
-}
-
 
-fn audit_package_rank_change(
-    req: &mut Request,
-    package_id: u64,
-    channel_id: u64,
-    operation: PackageChannelOperation,
-    origin_id: u64,
-) -> NetResult<NetOk> {
-    let mut audit = PackageChannelAudit::new();
-    audit.set_package_id(package_id);
-    audit.set_channel_id(channel_id);
-    audit.set_operation(operation);
+    pub fn channel_promote_package(req: &HttpRequest<AppState>) -> HttpResponse {
+        let (origin, channel) = Path::<(String, String)>::extract(&req)
+            .unwrap()
+            .into_inner();
 
-    let jgt = trigger_from_request(req);
-    audit.set_trigger(PackageChannelTrigger::from(jgt));
-
-    let (session_id, session_name) = get_session_id_and_name(req);
+        match Self::do_promote_package(req, origin, channel) {
+            Ok(_) => HttpResponse::new(StatusCode::OK),
+            Err(err) => err.into(),
+        }
+    }
 
-    audit.set_requester_id(session_id);
-    audit.set_requester_name(session_name);
-    audit.set_origin_id(origin_id);
+    pub fn channel_demote_package(req: &HttpRequest<AppState>) -> HttpResponse {
+        let (origin, channel) = Path::<(String, String)>::extract(&req)
+            .unwrap()
+            .into_inner();
+        match Self::do_demote_package(req, origin, channel) {
+            Ok(_) => HttpResponse::new(StatusCode::OK),
+            Err(err) => err.into(),
+        }
+    }
 
-    route_message::<PackageChannelAudit, NetOk>(req, &audit)
+    //
+    // Route registration
+    //
+    pub fn register(app: App<AppState>) -> App<AppState> {
+        app.resource("/channels/:origin", |r| {
```

Comment:
change the route registration params .. eg from ":origin" to "{origin}" (for example)
I missed that when wiring up some of the modules :)